File: Interop/cif/oneArray_cif.f90
program main

  ! Fortran program to call a module in C
  ! (file with the sabme name)
  
  use iso_c_binding
  implicit none

  integer, parameter :: N = 13 !Number of elements
  integer :: a(N), sum !vector and result

  data a(:)/N*1/ !Vector Values
  
  external FixVec !define external module

  call FixVec(N, a, sum) !Call for the module
  
  write(*,*) sum
  
  

end program main

File: Interop/cif/oneArray_cif.c
#include <stdio.h>

/* C module to be called from a fortran program */
/* in the file with the same name */


void fixvec_(int *N, int *v, int *sum){
  /* gets an array v (in) and an integer adress sum, calculates */
  /*   the sum of all the elements in v and writes it in sum */
  
  int i;
  *sum = 0;

  for (i = 0; i < *N; i++) {
    *sum = *sum + v[i];
    
  }

 
}
